packages/pyright-internal/src/analyzer/semanticTokensWalker.ts

```diff
@@ -147,8 +159,19 @@ export class SemanticTokensWalker extends ParseTreeWalker {
             case TypeCategory.Module:
                 this._addItem(node.start, node.length, SemanticTokenTypes.namespace, []);
                 return;
-            case TypeCategory.Unbound:
             case TypeCategory.Unknown:
+                // Check for untyped parameters
+                // eslint-disable-next-line no-case-declarations
+                const declarations = this._evaluator?.getDeclarationsForNameNode(node);
+                if (declarations?.some(isParameterDeclaration)) {
+                    const parent = declarations[0].node.parent as FunctionNode | LambdaNode;
+                    // Avoid duplicates for parameters visited by `visitParameter`
+                    if (!parent.parameters.some((param) => param.name?.id === node.id)) {
+                        this._addItem(node.start, node.length, SemanticTokenTypes.parameter, []);
+                    }
+                }
+                return;
```

Comment:
wrapping the case in a block will get rid of that eslint error. it will prevent the `declarations` variable from being in scope in other cases
```suggestion
            case TypeCategory.Unknown: {
                    // Check for untyped parameters
                    const declarations = this._evaluator?.getDeclarationsForNameNode(node);
                    if (declarations?.some(isParameterDeclaration)) {
                        const parent = declarations[0].node.parent as FunctionNode | LambdaNode;
                        // Avoid duplicates for parameters visited by `visitParameter`
                        if (!parent.parameters.some((param) => param.name?.id === node.id)) {
                            this._addItem(node.start, node.length, SemanticTokenTypes.parameter, []);
                        }
                    }
                }
                return;
```

Reply:
actually, since this same code is duplicated below on lines 223-229, can this one be turned into a `break` and instead be handled by that block below?
```suggestion
            case TypeCategory.Unknown:
                // handled below
                break;
```

Reply:
Yeah, the `break` would be better.